contracts/libraries/osmosis-cl-withdrawer/src/contract.rs

```diff
@@ -142,16 +164,30 @@ fn handle_liquidity_withdrawal_reply(
     let valence_callback = ValenceCallback::try_from(result)?;
 
     // decode the underlying position withdrawal response
-    // and query the pool to match the denoms
+    // and query the pool to ensure denom ordering
     let decoded_resp: MsgWithdrawPositionResponse = valence_callback.result.try_into()?;
     let pool = query_cl_pool(&deps, cfg.pool_id.u64())?;
 
+    let mut transfer_coins = vec![];
+
+    let amt_0 = Uint128::from_str(&decoded_resp.amount0)?;
+    let amt_1 = Uint128::from_str(&decoded_resp.amount1)?;
+
+    // there may be situations where only one coin was withdrawn.
+    // to avoid sending empty coins, we only include non-0-bal coins
+    if !amt_0.is_zero() {
+        transfer_coins.push(Coin::new(amt_0, pool.token0));
+    }
+    if !amt_1.is_zero() {
+        transfer_coins.push(Coin::new(amt_1, pool.token1));
+    }
+
+    // both coins cannot be zero because that would mean the position
+    // had no underlying liquidity to withdraw, so we skip the empty
+    // array check here and just fire the banksend
     let transfer_msg = BankMsg::Send {
         to_address: cfg.output_addr.to_string(),
-        amount: vec![
-            Coin::new(Uint128::from_str(&decoded_resp.amount0)?, pool.token0),
-            Coin::new(Uint128::from_str(&decoded_resp.amount1)?, pool.token1),
-        ],
+        amount: transfer_coins,
```

Comment:
Don't think for CL they are ordered btw, just looking at CL pools on osmosis and don't think they behave as GAMM. I also don't think it would matter as if it's not using proto, the bankmsg from wasm orders them, prob worth confirming it

Reply:
ah by ordering here I was referring to that we make sure that `amt_0` corresponds to `pool.token0` & same for `_1` to not mix up the denoms

Reply:
right